src/guards/AuthGuard.tsx

```
@@ -0,0 +1,22 @@
+import { useAuth } from "@/hooks/use-auth";
+import { getHistory, setHistory } from "@/utils/history";
+
+export const AuthGuard = ({ children }: { children: ReactNode }) => {
+  const { isAuthenticated } = useAuth();
+  const location = useLocation();
+
+  if (!isAuthenticated) {
+    return <Navigate to="/login" replace />;
+  }
+
+  useEffect(() => {
```

ESLint

[error] React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?

(react-hooks/rules-of-hooks)